src/Chapter3.hs

```diff
@@ -1006,7 +1122,78 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
-
+--Assuming data for each player (monster/knight)
+newtype PlayerHealth = PlayerHealth Int
+newtype PlayerAttack = PlayerAttack Int
+newtype PlayerDefense = PlayerDefense Int
+
+--All possible actions
+data BattleAction = Attack | DrinkPotion Int | CastSpell Int | MonsterAttack | MonsterRun
```

Comment:
It would be better to create two separate data types for `MonsterAction` and `FighterAction`, so it won't be possible to mix actions of different types :slightly_smiling_face: